numpy/random/_generator.pyx

```diff
@@ -4162,7 +4216,127 @@ cdef class Generator:
 
         return diric
 
-    # Shuffling and permutations:
+    def randomly_permute(self, object x, axis=None):
+        """
+        Randomly permute `x` along axis `axis` in-place.
+
+        If `axis` is None, the flattened array is shuffled.
+
+        Unlike `shuffle`, each slice along the given axis is shuffled
+        independently of the others.
+
+        Parameters
+        ----------
+        x : ndarray
+            Array to be randomly permuted.
+        axis : int, optional
+            Slices of `x` in this axis are permuted. Each slice
+            is permuted independently of the others.
+
+        Returns
+        -------
+        None
+
+        See Also
+        --------
+        numpy.random.Generator.randomly_permuted
```

Comment:
I think you can use short names since same object (i.e., remove numpy.random.Generator).

Reply:
Fixed.